### ISSUE TITLE
Cannot connect to local node

### ISSUE THREAD

**Issue by paradise** [bitpay/bitcore-node#565]
My problem is that bitcore-node cannot connect to my local Bitcoin ABC node. I've setup all like [here](https://github.com/bitpay/bitcore/issues/1471), but still getting errors:

```
[2018-05-19T12:45:56.836Z] info: Retrying connection to p2p network.
[2018-05-19T12:45:56.888Z] info: Disconnected from peer: 127.0.0.1
```

my bitcoin.conf file:
```
server=1
whitebind=127.0.0.1:8333
whitelist=127.0.0.1
txindex=1
addressindex=1
timestampindex=1
spentindex=1
zmqpubrawtx=tcp://127.0.0.1:28332
zmqpubrawtxlock=tcp://127.0.0.1:28332
zmqpubhashblock=tcp://127.0.0.1:28332
rpcallowip=127.0.0.1
rpcport=8332
```

my bitcore-node.json
```
{
  "version": "5.0.0-beta.44",
  "network": "livenet",
  "port": 3001,
  "services": [
    "address",
    "block",
    "fee",
    "db",
    "header",
    "mempool",
    "p2p",
    "timestamp",
    "transaction",
    "web",
    "insight-api"
  ],
  "datadir": "/ext/node/bitcoincash",
  "servicesConfig": {
    "insight-api": {
      "cwdRequirePath": "node_modules/insight-api"
    },
    "p2p": {
      "peers": [
        { "ip": { "v4": "127.0.0.1" } }
      ]
    }  
  }
}
```

what steps i've missed?

**Comment by paradise** [comment]
I've tested some cases. 
If I connect to public peer 0.16.x, then connection is successful, but this version does not support recent hard fork.
If I connect to public peer 0.17.x, then connection cannot be established. Error is the same as for a local node.

**Comment by eshangui** [comment]
join the question, I've met the same problem as you.

**Comment by lianhongHou** [comment]
try to use bitcore-lib-cash instead of bitcore-lib

**Comment by lianhongHou** [comment]
it is very slow to sync bitcore-node with bitcoincash0.17, which make bitcore-node useless

**Comment by linzhoushi2017** [comment]
@lianhongHou  can you tell me how to use bitcore-lib-cash instead of bitcore-lib?
This is my error. 
![_20180611195316](https://user-images.githubusercontent.com/29378230/41230036-dc847134-6db0-11e8-8647-9c37bab573a9.png)

**Comment by lianhongHou** [comment]
do not use bitcore-lib for bitcoincash, use bitcore-lib-cash instead, and replace 'require('bitcore-lib')' with 'require('bitcore-lib-cash')' in all js files in your project.

**Comment by osagga** [comment]
Yeah you can check this [fork](https://github.com/osagga/bitcore-node/tree/cash) of this node, it's only using `bitcore-lib-cash`, and it connects with a local bitcoin-abc 0.17.2.

**Comment by vladislavmamchich** [comment]
My problem is that bitcore-node cannot connect to my local Bitcoin ABC node (0.17.1). I've setup all like here, but still getting errors:

info: Retrying connection to p2p network.
info: Disconnected from peer: 127.0.0.1
